terraform/modules/frontdoor/cloudfront.tf

```diff
@@ -44,6 +45,31 @@ resource "aws_cloudfront_distribution" "main" {
     }
   }
 
+  origin {
+    domain_name = aws_s3_bucket.maintenance_page_bucket.website_endpoint
+    origin_id   = local.maintenance_origin_id
+
+    custom_origin_config {
+      http_port              = 80
+      https_port             = 443
+      origin_protocol_policy = "http-only"
+      origin_ssl_protocols   = ["TLSv1.2"]
+    }
+
+    custom_header {
+      name  = local.cloudfront_header_name
+      value = random_password.cloudfront_header.result
+    }
+  }
+
+  ordered_cache_behavior {
+    allowed_methods = ["GET", "HEAD"]
+    cached_methods = ["GET", "HEAD"]
+    path_pattern           = "/maintenance"
```

Comment:
This is required for an ordered_cache_behaviour - I think the idea is it will direct anything with the "/maintenance" path to this origin instead of the main origin.